### PR TITLE
Show summary info

### PR DESCRIPTION
**Pull request by adamrp**
There are a lot of commits here (sorry about that, was necessary to properly test and trouble-shoot on webdev).

If someone wants to sit down with me, I can review the changes in person if that would help.

### CODE REVIEW

**Comment by douginator2000** [comment]
Nice, thanks for factoring out the code into a new file. Portable is good. Is this deployed on webdev currently?

**Comment by adamrp** [comment]
It is -- but note that the histograms and sample grids will all be blank
(unless we copy some finished results from www over to webdev)

**Comment by douginator2000** [comment]
When I click on "Histograms" I get this error (selected study Arm_hand_sites
on webdev):

MOD_PYTHON ERROR

ProcessId:      13611
Interpreter:    'webdev.microbio.me'

ServerName:     'webdev.microbio.me'
DocumentRoot:   '/var/wwwroot/htdocs'

URI:            '/qiime/export_histograms.psp'
Location:       None
Directory:      '/var/wwwroot/htdocs/qiime/'
Filename:       '/var/wwwroot/htdocs/qiime/export_histograms.psp'
PathInfo:       ''

Phase:          'PythonHandler'
Handler:        'mod_python.psp'

Traceback (most recent call last):

  File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
line 1537, in HandlerDispatch
    default=default_handler, arg=req, silent=hlist.silent)

  File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
line 1229, in _process_target
    result = _execute_target(config, req, object, arg)

  File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
line 1128, in _execute_target
    result = object(arg)

  File "/usr/local/lib/python2.7/site-packages/mod_python/psp.py",
line 337, in handler
    p.run()

  File "/usr/local/lib/python2.7/site-packages/mod_python/psp.py",
line 243, in run
    exec code in global_scope

  File "/var/wwwroot/htdocs/qiime/export_histograms.psp", line 9, in <module>
    from summarize_seqs_otu_hits import histograms_as_html_table

ImportError: cannot import name histograms_as_html_table

On Mon, Apr 15, 2013 at 3:50 PM, adamrp notifications@github.com wrote:

> It is -- but note that the histograms and sample grids will all be blank
> (unless we copy some finished results from www over to webdev)
> 
> On Mon, Apr 15, 2013 at 3:47 PM, Doug Wendel notifications@github.comwrote:
> 
> > Nice, thanks for factoring out the code into a new file. Portable is
> > good.
> > Is this deployed on webdev currently?
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413605>
> > .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413734
> .

## 

Doug Wendel
Professional Research Associate
University of Colorado at Boulder
Department of Chemistry and Biochemistry, Knight Lab
303-641-5442
wendel@colorado.edu

**Comment by adamrp** [comment]
I saw that a few times, too, and I have no idea what's causing it.  The
function definitely exists (take a look at the file), but for some reason
it won't import it sometimes.  If you remove the .pyc file, it will
successfully import....  This also happens for the
utils.psp_utils.tab_delim_lines_to_table function.  Does anyone have any
ideas on that?

**Comment by ElDeveloper** [comment]
If you think that it would help we can go through the code together, just let me know when you want to work with this.

**Comment by adamrp** [comment]
Sure!  Tomorrow between 12:30 and 1 could work, or early (like 9)

**Comment by adamrp** [comment]
apachectl restart seemed to the trick... the code is still deployed on webdev right now, let me know if you see any errors like that again...

File: python_code/study_summary.py [hunk 1]
```diff
@@ -105,3 +105,72 @@ def print_study_info_and_values_table(query_results, data_access):
     
     return ''.join(info_table)
     
+def get_sample_summary_html(study_id, data_access):
+    out_string = ''
```
Comment (ElDeveloper) [inline python_code/study_summary.py:120]: :shipit: docstring?

File: python_code/summarize_seqs_otu_hits.py [hunk 2]
```diff
@@ -122,6 +123,12 @@ def summarize_all_stats(processed_dir):
     # Return all of the results
     return processed_results
 
+def histograms_as_html_table(processed_dir):
+    """
```
Comment (ElDeveloper) [inline python_code/summarize_seqs_otu_hits.py]: :shipit: docstring.

File: python_code/summarize_seqs_otu_hits.py [hunk 3]
```diff
@@ -122,6 +123,12 @@ def summarize_all_stats(processed_dir):
...
+    """
+    histograms_fp = join(processed_dir, 'split_libraries', 'histograms.txt')
+    return tab_delim_lines_to_table(open(histograms_fp).readlines())
```
Comment (ElDeveloper) [inline python_code/summarize_seqs_otu_hits.py]: U

File: python_code/tests/test_psp_utils.py [hunk 4]
```diff
@@ -37,5 +38,84 @@ def test_format_submit_form_to_fusebox_string(self):
         self.assertEqual(format_submit_form_to_fusebox_string(
             page='portal.psp'), change_page_only)
 
+    def test_tab_delim_lines_to_table(self):
```
Comment (ElDeveloper) [inline python_code/tests/test_psp_utils.py:41]: docstring

File: python_code/utils/psp_utils.py [hunk 5]
```diff
@@ -25,3 +25,29 @@ def format_submit_form_to_fusebox_string(**kwargs):
...
+    **kwargs: table properties that will be inserted directly into the HTML
+    <table> tag
+
+    will skip lines beginning with a pound sign (#)
+    """
+    table_opts = ' '.join('%s=%s' % (k,v) for (k,v) in kwargs.items())
+    table_str = '<table ' + table_opts + '>\n'
+
+    for line in lines:
+        if line and line[0] == '#':
```
Comment (ElDeveloper) [inline python_code/utils/psp_utils.py]: use startswith instead

File: www/export_histograms.psp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+from os.path import join
+from glob import glob
+from datetime import datetime
+import gzip
+from summarize_seqs_otu_hits import histograms_as_html_table
+
+sess = Session.Session(req)
+data_access = data_access_factory(ServerConfig.data_access_type)
+
+study_id = sess.get('study_id', form.get('study_id', ''))
```
Comment (ElDeveloper) [inline www/export_histograms.psp:19]: a comment?

File: python_code/study_summary.py [hunk 7]
```diff
@@ -105,3 +105,72 @@ def print_study_info_and_values_table(query_results, data_access):
...
+    sample_details = data_access.getSampleDetailList(study_id)
+    # Determine if we should write the seq and otu counts
+    write_seq_otu_cols = True
+    # This is the first sequence count field in the record. If None assume data
+    # does not yet exist for samples in general
+    if sample_details:
+        if sample_details[0][5] == None:
+            write_seq_otu_cols = False
+    else:
+        return 'This study has not yet been processed.'
```
Comment (ElDeveloper) [inline python_code/study_summary.py]: Different message here? Or no message? Suggestions @douginator2000 ?

**Comment by douginator2000** [comment]
The way the sample grid is handled on the study page is that if the study has not been processed, it still lists everything but the counts. So if you wanted to mirror the functionality then you can display that information.

**Comment by douginator2000** [comment]
And to clarify, it just omits the columns that would normally display the counts.

**Comment by adamrp** [comment]
Might be able to implement that, I'll look into that when I get a chance
later this week.

Adam

**Comment by adamrp** [comment]
It looks like it still displays everything, unless I'm misunderstanding something (see Agler_butyrate_from_corn_fiber, study id 741).

**Comment by adamrp** [comment]
This should be ready to go, unless there are further comments on the code.

**Comment by ElDeveloper** [comment]
This looks good. Mr. Señor Developer (a. k. a. @douginator2000) would you mind giving this a last check? Adam and I went through everything but a last check would be good :shipit:

**Comment by douginator2000** [comment]
Looks good to me!

**Pull request merged**